### PR TITLE
RavenDB-8521 Delayed external replication

### PR DESCRIPTION
**Pull request by karmeli87**
- Fixes after PR
- Check was added to send a heartbeat if we spend too much time in preparing a batch to send.

### CODE REVIEW

**Comment by karmeli87** [comment]
If we are skipping and suddenly interrupted, we will continue from the last point, since we keep the last etag.

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs [hunk 1]
```diff
@@ -212,8 +205,18 @@ public bool ExecuteReplicationOnce(OutgoingReplicationStatsScope stats, ref Date
                                     ((maxSizeToSend.HasValue && size > maxSizeToSend.Value.GetValue(SizeUnit.Bytes)) ||
                                      (batchSize.HasValue && numberOfItemsSent > batchSize.Value)))
                                 {
+                                    wasInterrupted = true;
                                     break;
                                 }
+
+                                if (_stats.Storage.CurrentStats.InputCount % 10000 == 0)
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:212]: use a power of 2 here.

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs [hunk 2]
```diff
@@ -293,6 +296,22 @@ public bool ExecuteReplicationOnce(OutgoingReplicationStatsScope stats, ref Date
...
 
+        private TimeSpan GetDelayReplication()
+        {
+            TimeSpan delayReplicationFor = TimeSpan.Zero;
+
+            if (_parent.Destination is ExternalReplication external)
+            {
+                delayReplicationFor = external.DelayReplicationFor;
+
+                if (delayReplicationFor.Ticks > 0 && _parent._parent._server.LicenseManager.CanDelayReplication(out var _) == false)
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:307]: where does the user get an error if the license doesn't allow it?

Comment (karmeli87) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:307]: It's validated inside the `CanDelayReplication` method and raise a notification if the license doesn't include this feature (treated as any other feature that is not included in the license).

**Pull request merged**